src/mesh.cpp

```diff
@@ -102,6 +121,70 @@ StructuredMesh::bin_label(int bin) const {
   }
 }
 
+//==============================================================================
+// Unstructured Mesh implementation
+//==============================================================================
+
+UnstructuredMesh::UnstructuredMesh(pugi::xml_node node) : Mesh(node) {
+  n_dimension_ = 3;
+
+  // check the mesh type
+  if (check_for_node(node, "type")) {
```

Comment:
Does this check need to be here? In `read_meshes(pugi::xml_node)`, we only build a `libMesh` or `moabMesh` if the node type is ``unstructured``:

```
 #ifdef DAGMC
     } else if (mesh_type == "unstructured" && mesh_lib == "moab") {
       model::meshes.push_back(std::make_unique<MOABMesh>(node));
 #endif
 #ifdef LIBMESH
     } else if (mesh_type == "unstructured" && mesh_lib == "libmesh") {
       model::meshes.push_back(std::make_unique<LibMesh>(node));
 #endif
```

Reply:
As the code stands, this is true. Long-term though, I don't mind having an extra check in the constructor. Having the check in the XML constructor ties the object creation to the XML spec we provide and insulates it from any exterior logic, which I like. Open to other thoughts about this though!

Reply:
That's fine with me! Just checking